.github/workflows/publish.yml

```diff
@@ -67,6 +67,9 @@ jobs:
         with:
           fetch-depth: 2
           ref: ${{ inputs.gitRef }}
+      - name: Update @o3r/pipeline package.json with git commit hash
+        run: |
+          jq '.o3rConfig.commitHash = "${{ github.sha }}"' packages/@o3r/pipeline/dist/package.json > tmp.json && mv tmp.json packages/@o3r/pipeline/dist/package.json
```

Comment:
Would be cleaner and more maintainable to be done in the `@o3r/pipeline` and potentially triggered by the nx task `publish`.
You can have access to the `GITHUB_SHA ` envvar as well from NX